### PR TITLE
migrate test infra from cdk v1 to cdk v2

### PR DESCRIPTION
**Pull request by cnfait**
### Feature or Bugfix
- Refactoring

### Detail
- migrate test infra from cdk v1 to cdk v2

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 665949838a0498457fd799a4d2d0dd7d690f2a20
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=827ced0e-f48b-4441-b401-6302906cacf7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: test_infra/stacks/databases_stack.py [hunk 1]
```diff
@@ -475,13 +477,13 @@ def _setup_mysql_serverless(self) -> None:
...
-                auto_pause=cdk.Duration.minutes(5),
+                auto_pause=Duration.minutes(5),
                 min_capacity=rds.AuroraCapacityUnit.ACU_1,
                 max_capacity=rds.AuroraCapacityUnit.ACU_1,
             ),
-            backup_retention=cdk.Duration.days(1),
+            backup_retention=Duration.days(1),
             vpc=self.vpc,
-            vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PRIVATE),
+            vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PRIVATE_WITH_NAT),
```
Comment (kukushking) [inline test_infra/stacks/databases_stack.py:486]: Just checking if that is intended difference?

Comment (kukushking) [inline test_infra/stacks/databases_stack.py:486]: Just checking if that is intended difference?

File: test_infra/cdk.json [hunk 2]
```diff
@@ -1,17 +1,6 @@
 {
```
[the remark this answers is not in the document]
Comment (cnfait) [reply test_infra/cdk.json:1]: Good suggestion, I'll make the change!

[the remark this answers is not in the document]
Comment (cnfait) [reply test_infra/cdk.json:1]: Good suggestion, I'll make the change!

File: test_infra/stacks/databases_stack.py (hunk 1 above)
Comment (cnfait) [reply test_infra/stacks/databases_stack.py:486]: Yes, `ec2.SubnetType.PRIVATE` does not exist in cdk v2 and [is marked as deprecated in CDK v1.](https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.aws_ec2/SubnetType.html#aws_cdk.aws_ec2.SubnetType.PRIVATE)

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 6634346d77b7ec7f61cfaea130f96997d4244c2f
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c74e2aa5-026c-4835-96bb-625e32b7d1a4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]
LGTM!

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 67ba36f3b29ede5fd2d6b9f5d091a8b2a32d03cd
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d6a2b89c-9700-429d-90dc-b7efe1a21cc4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**